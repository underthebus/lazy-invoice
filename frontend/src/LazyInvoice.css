.ub-LazyInvoice {
  padding: 1rem;
}
