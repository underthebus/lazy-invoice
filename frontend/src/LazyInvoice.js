import { hot } from 'react-hot-loader/root';
import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { format } from 'date-fns';

const GET_INVOICES = gql`
      query GetInvoices {
        invoices {
          date
          items {
            description
            quantity
            unitPrice
          }
        }
      }
`;

export const Component = () =>
  <div>
    <div data-e2e="example">
      Hello Feibian!
    </div>
    <Query query={GET_INVOICES}>
      {({ data, loading, error }) => {
        if (loading) return <div>Loading ...</div>;
        if (error) return <div>Error!</div>;

        return (
          <div>
            {data.invoices.map((invoice, i) =>
              <div key={i}>
                {`Invoice date: ${format(invoice.date, 'd MMMM, YYYY' )}`}
                {invoice.items.map((item, i) =>
                  <div key={i} style={{ display: 'flex'}}>
                    <div style={{ width: 150 }}>{item.description}</div>
                    <div style={{ width: 50 }}>{item.quantity}</div>
                    <div style={{ width: 50 }}>{item.unitPrice}</div>
                  </div>
                )}
              </div>)}
          </div>
        );
      }}
    </Query>
  </div>;

export default hot(Component);
