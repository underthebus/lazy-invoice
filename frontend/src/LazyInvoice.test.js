import React from 'react';
import { shallow } from 'enzyme';

import { Component } from './LazyInvoice';

it('renders as expected', () => {
  const wrapper = shallow(<Component />);
  expect(wrapper).toMatchSnapshot();
});
