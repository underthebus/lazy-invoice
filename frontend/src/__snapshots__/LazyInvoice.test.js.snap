// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "GetInvoices",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "invoices",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "date",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "items",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "description",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "quantity",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "unitPrice",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [
                    Object {
                      "arguments": Array [],
                      "kind": "Directive",
                      "name": Object {
                        "kind": "Name",
                        "value": "client",
                      },
                    },
                  ],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "user",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 224,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
</div>
`;
